Resolves Issue/PR templates

ve-varun-sharma opened the pull request:
## Pull Request Type

Dictionary of Definitions here:
https://ben.balter.com/2015/12/08/types-of-pull-requests/

- [x] Just a heads up 🙆
- [ ] Sanity check 🏃‍♂️
- [ ] Work in progress (WIP) 🏗️
- [ ] Early feedback ✍️
- [ ] Line-by-line review 📝
- [ ] Pull request to a pull request 🔀

## Description

The PR offers to add some additional documentation structure for github issues in the repo and PRs for smoother review process.
If this is unneeded or feels too much structure no worries at all! I thought this may help with improved clarity, faster resolution, and increase better collaboration via providing sufficient details!

When users go to the Github UI to create a Github Issue or Pull Request, the new templates in the .github folder will automatically pull in the templates ensuring that users follow the desired instructions in the templates.

This PR addresses this Github issue: https://github.com/DefangLabs/defang/issues/705


 

#### Caveats/Notes (optional)

This PR doesn't touch any direct code so this will not affect an code and thus features.


#### Prompts (optional)

Please feel free to iterate, add, edit, delete from the proposed templates.

### Screenshots (optional)

Not needed for this.

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [x] If it is a core feature, I have added thorough tests.
- [x] Extended the README / documentation, if necessary
- [x] My changes generate no new warnings
- [x] Is the code readable? Prefer to see more self-explanatory and consistency
- [x] Is the code tested properly? i.e. where appropriate, we're testing well-explained failures, behaviours, fixes, etc

lionello commented:
Thanks! Assigning to @jordanstephens for review, because he has done some work on this before.

ve-varun-sharma commented:
> This is cool! Can we make it more concise though? Too much boilerplate makes it hard to find the meat of the issue/pr.
> 
> "This PR addresses this Github issue" doesn't trigger GitHub to link the issue to the PR, see https://docs.github.com/en/issues/tracking-your-work-with-issues/using-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword. Wonder if that could be part of the template?

Ah yes of course!  I can certainly utilize the key word triggering there.

on the template, definitely. Are there parts that you like/dislike? What would you ideally like to have in there? I can help make the edits :)

ve-varun-sharma commented:
> This is cool! Can we make it more concise though? Too much boilerplate makes it hard to find the meat of the issue/pr.
> 
> "This PR addresses this Github issue" doesn't trigger GitHub to link the issue to the PR, see https://docs.github.com/en/issues/tracking-your-work-with-issues/using-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword. Wonder if that could be part of the template?

@lionello 
I implemented your feedback, cut out some sections, and added a section based on what you suggested! Let me know if there's anything else we want to update this with :)

Ready for review! 
tagging @jordanstephens  for visibility here too

jordanstephens reviewed (approved):
Thanks for submitting these templates. I appreciate the spirit of these changes, though I think these templates are a bit heavy for our current stage. I've made some suggestions inline to tighten them up for our use-case.

ve-varun-sharma commented on .github/ISSUE_TEMPLATE.md:
```diff
@@ -0,0 +1,27 @@
+- Version of Defang or relevant software.
+- Any relevant dependencies.
+
+## Screenshots/Logs:
+
+- Attach any relevant screenshots, error messages, or log files.
+
+## Additional Information:
```
agreed!

ve-varun-sharma commented on .github/PULL_REQUEST_TEMPLATE.md:
```diff
@@ -0,0 +1,52 @@
+## Description
+
+- Concise description of what this PR is tackling.
+
+## Linked Issues
+
+This section will automatically populate with linked issues.
+
+[To link an issue to this pull request, use one of the following keywords linked here in the pull request description](https://docs.github.com/en/issues/tracking-your-work-with-issues/using-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword):
+
+- `close`
+- `closes`
+- `closed`
+- `fix`
+- `fixes`
+- `fixed`
+- `resolve`
+- `resolves`
+- `resolved`
+
+For example:
+
+- `Closes #10`
+- `Fixes octo-org/octo-repo#100`
+- `Resolves #10, resolves #123, resolves octo-org/octo-repo#100`
+
+### Screenshots (optional)
+
+[Upload optional screenshots]
+
+## Checklist before requesting a review
+
+- [ ] I have performed a self-review of my code
+- [ ] If it is a core feature, I have added thorough tests.
+- [ ] Extended the README / documentation, if necessary
+- [ ] My changes generate no new warnings
+- [ ] Is the code readable? Prefer to see more self-explanatory and consistency
+- [ ] Is the code tested properly? i.e. where appropriate, we're testing well-explained failures, behaviours, fixes, etc
+
+##### Memes Are welcomed
```
Good callout here! I agree with your line of thinking. I like the simpler approach for this too

ve-varun-sharma commented on .github/ISSUE_TEMPLATE.md:
```diff
@@ -0,0 +1,27 @@
+
+## Environment:
+
+- Operating system (e.g., Windows, macOS, Linux).
+- Browser (if applicable).
+- Version of Defang or relevant software.
+- Any relevant dependencies.
```
Nice changes to this too!

lionello commented on .github/PULL_REQUEST_TEMPLATE.md:
```diff
@@ -0,0 +1,15 @@
+
```
```suggestion
```

lionello commented on .github/PULL_REQUEST_TEMPLATE.md:
```diff
@@ -0,0 +1,15 @@
+
+- [ ] I have performed a self-review of my code
+- [ ] I have added appropriate tests
+- [ ] I have updated the Defang CLI help-text to reflect my changes, if necessary
```
```suggestion
- [ ] I have updated the Defang CLI docs and/or README to reflect my changes, if necessary
```

lionello commented on .github/ISSUE_TEMPLATE.md:
```diff
@@ -0,0 +1,20 @@
+
+## Additional Information, Relevant dependencies, Screenshots, Logs, Etc...
+
+Any other relevant details that might be helpful.
```
```suggestion
<!--Any other relevant details that might be helpful. -->
```

lionello reviewed (approved):
LGTM
